metrics/cli/docs/features/update/06_update_feature.md

```diff
@@ -0,0 +1,139 @@
+
+### Feasibility study
+> A preliminary study of the feasibility of implementing this feature.
+
+The `Update feature` expands the end-user capabilities and improves the UX of the Metrics CLI, so this feature
+makes sense for the project.
+
+Moving on to the feasibility of implementation, the `Update feature` requires the following actions:
+
+- parsing the YAML configuration file;
+- redeploying/updating the existing Metrics Web application to the same Firebase project.
+
+The following proofs cover the above requirements, so we can implement the `Update feature`:
+- to parse the YAML configuration file, we can use a [package](https://github.com/Flank/flank-dashboard/tree/master/yaml_map) that applies in the ci_integrations module for the same purposes;
+- the Metrics CLI has everything we need to perform the redeploy action to the same Firebase project since the CLI can perform a deploy action.
+
+### Requirements
+> Define requirements and make sure that they are complete.
+
+Let's start with visiting and analyzing resources to collect the necessary data during the analysis stage.
+Thanks to these data, we can create precise and determined requirements and edge cases for the `Update feature`.
+
+The `Update feature` has the following requirements:
+- should read the configuration from the YAML configuration file;
+- should consume the configuration file path via the command line arguments;
+- should validate the YAML configuration file;
+- should stop the redeploy process of the Metrics CLI if something went wrong during the process;
+- should redeploy/update the existing Metrics Web application to the same Firebase project;
+- configuring the Sentry should be optional.
+
+### Landscape
+> Look for existing solutions in the area.
+
+There are no ready packages or solutions that satisfy our requirements since the `Update feature` is a custom feature 
+necessary exclusively for Metrics specific purposes in the Metrics CLI tool. Considering all the previous, custom implementation is the best choice.
+
+
+Let's take a closer look at the custom implementation.
+
+This approach means we should implement the command, which redeploys the Metrics Web app to the existing Firebase
+project. The command should extract the required data for the redeploy from the YAML configuration file.
+
+### Prototyping
+> Create a simple prototype to confirm that implementing this feature is possible.
+
+As we've described in the requirements section, we must be able to perform the following actions to implement the Update
+feature:
+
+- should read the configuration from the YAML configuration file;
+- should consume the configuration file path via the command line arguments;
+- should validate the YAML configuration file;
+- should stop the redeploy process of the Metrics CLI if something went wrong during the process;
+- should redeploy/update the existing Metrics Web application to the same Firebase project;
+- configuring the Sentry should be optional.
```

Comment:
It seems like we duplicate the requirements section here. Not sure we should do something like this in the Prototyping section. Write the natural text here starting from the short intro.

```suggestion
To confirm that we are able to implement the update feature, let's start with configuration file parsing. The configuration file is a YAML-formatted file that defines the update targets (e.g., Firebase project). We can use Dart's core File implementation to read the configuration file as follows: 
... 

Then, having the configuration file content in a string, we can parse it using the yaml_map:
... 

The resulting map contains the values from the configuration file:
...
```
And so on.